docs/user/device-api-statuses.md

```diff
@@ -0,0 +1,237 @@
+
+The `device.status.updated.status` field can have the following values:
+
+| Status | Description | Formal Definition<sup>1</sub> |
+| ------ | ----------- | ----------------------------- |
+| `UpToDate` | The device is updated to its device spec. If the device is member of a fleet, its device spec is at the same template version as its fleet's device template. | `!deviceIsUpdating && deviceIsUpdatedToDeviceSpec && (deviceIsNotManaged \|\| deviceIsUpdatedToFleetSpec)` |
+| `Updating` | The device is in the process of updating to its device spec. | `deviceIsUpdating` |
+| `OutOfDate` | The device is not updating and either not updated to its device spec or - if it is member of a fleet - its spec is not yet of the same template version as its fleet's device template. | `!deviceIsUpdating && (!deviceIsUpdatedToDeviceSpec \|\| (deviceIsManaged && !deviceIsUpdatedToFleetSpec))` |
+| `Unknown` | The device's agent either never reported status or its last reported status was `Updating` and the device has been disconnected since. | `deviceIsDisconnected && lastStatus == Updating` |
+
+<sup>1</sup> For the detailed definitions derived from the device specs and statuses, see [Helper Definitions](#helper-definitions).
+
+The `device.status.updated.???` field contains the current state of the update in progress and can take the following values:
+
+| Update State | Description |
+| ------------ | ----------- |
+| `Preparing` | The agent is validating the desired device spec and downloading dependencies. No changes have been made to the device's configuration yet. |
+| `ReadyToUpdate` | The agent has validated the desired spec, downloaded all dependencies, and is ready to update. No changes have been made to the device's configuration yet. |
+| `ApplyingUpdate` | The agent has started the update transaction and is writing the update to disk. |
+| `Rebooting` | The agent initiated a reboot required to activate the new OS image and configuration. |
+| `ActivatingConfig` | (transient, not reported) The agent is activating the new configuration without requiring a reboot. |
+| `RollingBack` | The agent has detected an error and is rolling back to the pre-update OS image and configuration. |
+| `Updated` | The agent has successfully completed the update and the device is conforming to its device spec. Note that the device's update status may still be reported as `OutOfDate` if the device spec is not yet at the same version as the fleet's device template. |
+| `Canceled` | The agent has canceled the update because the desired spec was reverted to the current spec before the update process started. **[TODO: Do we keep downloaded dependencies?]** |
+| `Error` | The agent failed to apply the desired spec and will not retry. The device's OS image and configuration have been rolled back to the pre-update version and have been activated. **[TODO: Do we keep downloaded dependencies?]** |
```

Comment:
In the case that the agent can retry I currently have a `Retry`

Reply:
`Retry` doesn't exist in the diagram below and I'm not sure where we'd add this state and transitions. Would it even be a separate state rather than a transition back into the same or a previous state?

`Error` as state is important, because we need to signal to users that the update failed finally (i.e. after max retries) and so this version of the spec will never be reconciled again.

Reply:
> Could you describe a scenario in which an update would fail but be successful a second time?
An example would networking failure during image prefetch. In this case we have hit a retryable error.

Reply:
Sure I guess I was thinking the user would like to know that the agent failed for a retryable error. But maybe that is just bug food and the user should not need to know about retry.

Reply:
we can drop the update in the agent.